### PR TITLE
Created support for example keyword in generatedExample

### PR DESCRIPTION
**Pull request by Danielecina**
#### Checklist

- [x] your branch will not cause merge conflict with `master`
- [x] run `npm test`
- [x] tests are included
- [ ] the documentation is updated or integrated accordingly with your changes
- [x] the code will follow the lint rules and style of the project
- [x] you are not committing extraneous files or sensible data

### CODE REVIEW

File: example/swagger-files/anyOf-example.json [hunk 1]
```diff
@@ -123,7 +133,7 @@
     "schemas": {},
     "responses": {},
     "parameters": {},
-    "examples": {},
+    "example": {},
```
Comment (Danielecina) [inline example/swagger-files/anyOf-example.json:136]:
```suggestion
    "examples": {},
```

File: packages/api-explorer/package.json [hunk 2]
```diff
@@ -35,7 +35,7 @@
   },
   "scripts": {
     "lint": "eslint -f unix . --ext js --ext jsx",
-    "inspect": "jsinspect",
+    "inspect": "jsinspect -t 50",
```
Comment (Danielecina) [inline packages/api-explorer/package.json:38]: reduced the threshold of identical pieces of code

File: example/swagger-files/anyOf-example.json [hunk 3]
```diff
@@ -21,13 +21,20 @@
...
                     "properties": {
-                      "pet_type": {
+                      "_id": {
+                        "pattern": "^[a-fA-F\\d]{24}$",
+                        "example": "000000000000000000000000",
                         "type": "string"
                       },
+                      "pet_type": {
+                        "type": "string",
+                        "example": "bassotto"
```
Comment (fredmaggiowski) [inline example/swagger-files/anyOf-example.json:31]:
```suggestion
                        "example": "dachshund"
```

maybe in english could be better :)

File: example/swagger-files/anyOf-example.json [hunk 4]
```diff
@@ -21,13 +21,20 @@
...
+                      },
                       "pet_children": {
                         "type": "array",
                         "items": {
-                          "type": "string"
+                          "type": "string",
+                          "example": "bassotto 2"
```
Comment (fredmaggiowski) [inline example/swagger-files/anyOf-example.json:37]:
```suggestion
                          "example": "long haired dachshund"
```

File: example/swagger-files/anyOf-example.json [hunk 5]
```diff
@@ -37,7 +44,8 @@
                     "type": "object",
                     "properties": {
                       "pet_name": {
-                        "type": "string"
+                        "type": "string",
+                        "example": "bassotto"
```
Comment (fredmaggiowski) [inline example/swagger-files/anyOf-example.json:48]:
```suggestion
                        "example": "dachshund"
```

File: example/swagger-files/anyOf-example.json [hunk 6]
```diff
@@ -46,10 +54,12 @@
                     "type": "object",
                     "properties": {
                       "pet_type": {
-                        "type": "string"
+                        "type": "string",
+                        "example": "bassotto"
```
Comment (fredmaggiowski) [inline example/swagger-files/anyOf-example.json:58]:
```suggestion
                        "example": "dachshund"
```

File: example/swagger-files/anyOf-example.json [hunk 7]
```diff
@@ -46,10 +54,12 @@
...
                       },
                       "pet_country": {
-                        "type": "string"
+                        "type": "string",
+                        "example": "spagna"
```
Comment (fredmaggiowski) [inline example/swagger-files/anyOf-example.json:62]:
```suggestion
                        "example": "spain"
```

File: packages/api-explorer/lib/generateFakeSchema.js [hunk 8]
```diff
@@ -8,6 +8,37 @@ jsf.option({
   alwaysFakeOptionals: true,
 })
 
+function generateObjectExample(example, properties) {
+  const generatedExample = { ...properties ? jsf.generate(properties) : {} }
```
Comment (fredmaggiowski) [inline packages/api-explorer/lib/generateFakeSchema.js:12]: Why the spread operator?

```suggestion
  const generatedExample = properties ? jsf.generate(properties) : {}
```

File: packages/api-explorer/lib/generateFakeSchema.js [hunk 9]
```diff
@@ -8,6 +8,37 @@ jsf.option({
...
+  // eslint-disable-next-line guard-for-in
```
Comment (fredmaggiowski) [inline packages/api-explorer/lib/generateFakeSchema.js:13]: instead of disabling the lint is there another approach available for this iteration? For instance an Object.entries().forEach?

File: packages/api-explorer/lib/generateFakeSchema.js [hunk 10]
```diff
@@ -8,6 +8,37 @@ jsf.option({
...
+  for (const key in example) {
+    generatedExample[key] = generateExampleFromType(example[key], example[key])
+  }
+  return generatedExample
+}
+
+function generateArrayExample(example) {
+  const generatedExample = []
```
Comment (fredmaggiowski) [inline packages/api-explorer/lib/generateFakeSchema.js:21]: `return example.map(item => generatedExample.push(generateExampleFromType(item, item))` ?

File: packages/api-explorer/lib/generateFakeSchema.js [hunk 11]
```diff
@@ -8,6 +8,37 @@ jsf.option({
...
+    generatedExample.push(generateExampleFromType(item, item))
+  })
+  return generatedExample
+}
+
+function generateExampleFromType (example, schema) {
+  if(Array.isArray(example)) {
+    return generateArrayExample(example)
+  }
+  if (example !== null && !Array.isArray(example) && typeof example === 'object'){
```
Comment (fredmaggiowski) [inline packages/api-explorer/lib/generateFakeSchema.js:32]:
```suggestion
  if (example !== null && !Array.isArray(example) && typeof example === 'object') {
```

**Review by fredmaggiowski — CHANGES_REQUESTED** [review]
I've put a few comments on code style

File: packages/api-explorer/lib/generateFakeSchema.js (hunk 10 above)
Comment (Danielecina) [reply packages/api-explorer/lib/generateFakeSchema.js]: done `example.map(item => generateExampleFromType(item, item))`

File: packages/api-explorer/lib/generateFakeSchema.js (hunk 9 above)
Comment (Danielecina) [reply packages/api-explorer/lib/generateFakeSchema.js]: I turned "off" the `guard-for-in` in .eslintrc. If i use other method i call two cycle to get this information.
This linting is not so useful. If i choose to exclude some key from cycle i can do it without his constraint. What do you think?

File: packages/api-explorer/lib/generateFakeSchema.js [hunk 12]
```diff
@@ -9,27 +9,22 @@ jsf.option({
...
   return generatedExample
 }
 
 function generateArrayExample(example) {
-  const generatedExample = []
-  example.forEach(item => {
-    generatedExample.push(generateExampleFromType(item, item))
-  })
-  return generatedExample
+  return example.map(item => generateExampleFromType(item, item))
```
Comment (fredmaggiowski) [inline packages/api-explorer/lib/generateFakeSchema.js:20]: Why does this take the same argument twice?

Comment (Danielecina) [reply packages/api-explorer/lib/generateFakeSchema.js:20]: The listener on the example key, gives me the value of it and the rootSchema it belongs to
If the example key is a primitive type as a string, we return it directly.
If the example key is a structural type we need to check recursive the children.
In this case:
- The first parameter is used to understand the `type`
- The second is only useful if it were an object to get hypothetical `properties`. We repeat twice item word as parameter because we don't know if it's an object.

Comment (Danielecina) [reply packages/api-explorer/lib/generateFakeSchema.js:20]: This test help to understand the recursion
```js
{
      title: 'use example outside properties in array of object',
      schema: {
        type: 'object',
        required: ['foo'],
        example: {foo: [{fooChildren: 'nestedBar'}]},
        properties: {
          foo: {
            type: 'array',
            minItems: 1,
            maxItems: 1,
            items: {
              type: 'object',
              properties: {
                fooChildren: {
                  type: 'string',
                }
              }
            }
          },
          randomKey: {
            type: 'string'
          }
        }
      },
      expectation: generatedFakeSchema => {
        expect(generatedFakeSchema).toEqual({foo: [{fooChildren: 'nestedBar'}], randomKey: expect.any(String)})
      }
```

**Review by fredmaggiowski — APPROVED** [review]

File: packages/api-explorer/__tests__/JsonForm/__snapshots__/index.test.jsx.snap [hunk 13]
```diff
@@ -44,14 +44,13 @@ exports[`JSONForm  snapshot 1`] = `
     type="submit"
   />
   <div
-    class="mia-container-wrapper"
+    class="je-object__container mia-container-wrapper"
```
Comment (dibericky) [inline packages/api-explorer/__tests__/JsonForm/__snapshots__/index.test.jsx.snap:47]: Why these changes?

Comment (Danielecina) [reply packages/api-explorer/__tests__/JsonForm/__snapshots__/index.test.jsx.snap:47]: there shouldn't be these changes in fact. nothing was touched

Comment (Danielecina) [reply packages/api-explorer/__tests__/JsonForm/__snapshots__/index.test.jsx.snap:47]: I try to understand as soon as I can

File: packages/api-explorer/lib/generateFakeSchema.js (hunk 12 above)
Comment (fredmaggiowski) [reply packages/api-explorer/lib/generateFakeSchema.js:20]: I did not properly understand it but ok, it just seems weird to me why we need two arguments

**Review by fredmaggiowski — APPROVED** [review]

**Review by fredmaggiowski — APPROVED** [review]
